widget/embedded/src/components/SwapDetails/SwapDetails.helpers.tsx

```diff
@@ -0,0 +1,215 @@
+export const PERCENT_DECIMALS = 2;
+export const SECONDS = 60;
+
+export const getSteps = (
+  pendingSwap: PendingSwap,
+  switchNetwork: (() => Promise<ConnectResult>) | undefined,
+  showNetworkModal: PendingSwapNetworkStatus | null | undefined,
+  setNetworkModal: (network: PendingSwapNetworkStatus) => void,
+  message: ReturnType<typeof getSwapMessages>
+): Step[] => {
+  return pendingSwap.steps.map((step, index) => {
+    return {
+      from: {
+        token: { displayName: step.fromSymbol, image: step.fromLogo },
+        chain: {
+          displayName: step.fromBlockchain,
+          image: step.fromBlockchainLogo,
+        },
+        price: {
+          value:
+            index === 0
+              ? numberToString(
+                  pendingSwap.inputAmount,
+                  AMOUNT_DECIMALS,
+                  AMOUNT_DECIMALS
+                )
+              : numberToString(
+                  pendingSwap.steps[index - 1].outputAmount,
+                  AMOUNT_DECIMALS,
+                  AMOUNT_DECIMALS
+                ),
+        },
+      },
+      to: {
+        token: { displayName: step.toSymbol, image: step.toLogo },
+        chain: {
+          displayName: step.toBlockchain,
+          image: step.toBlockchainLogo,
+        },
+        price: {
+          value: step.outputAmount
+            ? numberToString(
+                step.outputAmount,
+                AMOUNT_DECIMALS,
+                AMOUNT_DECIMALS
+              )
+            : numberToString(
+                step.expectedOutputAmountHumanReadable,
+                AMOUNT_DECIMALS,
+                AMOUNT_DECIMALS
+              ),
+        },
+      },
+      swapper: { displayName: step.swapperId, image: step.swapperLogo },
+      alerts: (
+        <Alerts>
+          {step.explorerUrl?.map((explorerUrl, index, urls) => {
+            const key = index + explorerUrl.url;
+            return (
+              <Fragment key={key}>
+                <Alert
+                  type={
+                    (index === urls.length - 1 &&
+                      step.status === 'running' &&
+                      (index !== 0 ||
+                        (index === 0 &&
+                          explorerUrl.description !== 'Approve'))) ||
+                    (step.status === 'waitingForApproval' &&
+                      !isNetworkStatusInWarningState(step))
+                      ? 'loading'
+                      : index === urls.length - 1 &&
+                        (index !== 0 ||
+                          (index === 0 &&
+                            explorerUrl.description !== 'Approve')) &&
+                        step.status === 'failed'
+                      ? 'error'
+                      : 'success'
+                  }
+                  title={
+                    !explorerUrl.description
+                      ? i18n.t('View transaction')
+                      : `${explorerUrl.description
+                          .substring(0, 1)
```

Comment:
Maybe we could extract a capitalize function and use it here.

Reply:
fixed.